data-prepper-core/src/main/java/org/opensearch/dataprepper/event/DefaultEventHandle.java

```diff
@@ -6,6 +6,16 @@
 package org.opensearch.dataprepper.event;
 
 import org.opensearch.dataprepper.model.event.EventHandle;
+import org.opensearch.dataprepper.model.acknowledgements.AcknowledgementSet;
+import java.lang.ref.WeakReference;
 
-class DefaultEventHandle implements EventHandle {
+public class DefaultEventHandle implements EventHandle {
+    private final WeakReference acknowledgementSetRef;
```

Comment:
`WeakReference` supports generics. So you can use `WeakReference<AcknowledgementSet>` as the type.